vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java

```diff
@@ -483,32 +466,69 @@ protected String promptForSide() {
     }
 
     availableSides.removeAll(alreadyTaken);
+
+    // If a "real" player side is available, we want to offer "the next one" as the default, rather than observer.
+    // Thus hotseat players can easily cycle through the player positions as they will appear successively as the default.
+    // Common names for Solitaire players (Solitaire, Solo, Referee) do not count as "real" player sides, and will be skipped.
+    // If we have no "next" side available to offer, we stay with the observer side as our default offering.
+    boolean found = false;       // Haven't found our own side in the list.
```

Comment:
What if the only available sides are _before_ the side we have now?

Reply:
See line 499 down below the if block.

Reply:
Ok.

So the rule is: Select the first available non-solo side after the current one, wrapping around to the head of the list if necessary?

Reply:
Correct! A "modulo" next-available-non-solo-side, if you will. So imagine you're in the hotseat use case, then whenever you click the "resign" button it comes up with the next logical side for you, so you can basically just click resign button and hit Enter and you'll be on the next side (or hit the Resign Hotkey and then Enter). In a two-player game it will just swap you back and forth between the two sides. Obviously at any point you can still manually pick any of the other sides, or "Observer".

Reply:
Makes sense. That's a good rule.